src/pybind/mgr/dashboard/controllers/orchestrator.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+
+import cherrypy
+
+from . import ApiController, Endpoint, ReadPermission
+from . import RESTController, Task
+from ..security import Scope
+from ..services.orchestrator import OrchClient
+from ..tools import wraps
+
+
+def orchestrator_task(name, metadata, wait_for=2.0):
+    return Task("orchestrator/{}".format(name), metadata, wait_for)
+
+
+def raise_if_no_orchestrator(method):
+    @wraps(method)
+    def inner(self, *args, **kwargs):
+        orch = OrchClient.instance()
+        if not orch.available():
+            raise cherrypy.HTTPError(503)
+        return method(self, *args, **kwargs)
+    return inner
+
+
+@ApiController('/orchestrator')
+class Orchestrator(RESTController):
+
+    @Endpoint()
+    @ReadPermission
+    def status(self):
+        return OrchClient.instance().status()
+
+
+@ApiController('/orchestrator/inventory', Scope.HOSTS)
+class OrchestratorInventory(RESTController):
+
+    @raise_if_no_orchestrator
+    def list(self, hostname=None):
+        orch = OrchClient.instance()
+        hosts = [hostname] if hostname else None
+        inventory_nodes = orch.inventory.list(hosts)
+        return [node.to_json() for node in inventory_nodes]
+
+
+@ApiController('/orchestrator/service', Scope.HOSTS)
+class OrchestratorService(RESTController):
+
+    @raise_if_no_orchestrator
+    def list(self, hostname=None):
+        orch = OrchClient.instance()
+        return [service.to_json() for service in orch.services.list(None, None, hostname)]
```

Comment:
Does this overlap with the OSD list? I wonder if it would make sense to somehow unify the view of the orchestrator's services with the rest of the dashboard?

Reply:
Yeah, Orchestrator service entities should be hidden behind various components of Dashboard. We already have an OSD list and if we want to manipulate OSD, we then just leverage orchestrator API calls in the backend. This also applies to MON, MDS, ..etc.

We don't have an issue on the tracker to display all orchestrator services (but we have one pending item on the [etherpad](https://pad.ceph.com/p/ceph-dashboard-octopus-priorities) during f2f, item 22.6). I implemented it just because orchestrator CLI has it.

Do users have the need to see *all* services or services of a host? Also, orchestrator services display slightly more information. For example, when using a Rook backend it might be useful to know the container id. Of course, this can also be done by showing additional info on MON/MDS/MON components too.